packages/utils/src/useTimeout.ts

```diff
@@ -24,6 +24,19 @@ export class Timeout {
     }, delay) as unknown as number; /* Node.js types are enabled in development */
   }
 
+  /**
+   * Executes `fn` after `delay` but only if no other timeout is ongoing.
+   */
+  startIfEmpty(delay: number, fn: Function) {
```

Comment:
`startIfNotPending`?

Reply:
Imo, doesn't sound better than `.startIfEmpty()`, although I'm not very satisfied with either name. I also thought about naming it `.throttle()`, although I've always been confused by "debounce" and "throttle" so I picked a name that describes directly what the function does.

We also have the `.isStarted()` method, it would be nice to tie it all together into two states ("empty" and "...."?) rather than use different terms ("started" and "pending").

Open to more suggestions/feedback.

Reply:
"pending" is a term already used in web animation APIs (along with "started", "idle", etc.) and it's pretty clear. "empty" is not intuitive for me at all (I'm not "filling" anything when setting a timeout).

Reply:
The `Timeout` instance can only hold a single `setTimeout` instance at once, that's what's being filled. The base method is also named `.start()` instead of `.set()` because it's not just a `setTimeout`, it also clears the pending `setTimeout` if it already holds one.

I guess I'm ok with "pending", but then I will rename `isStarted` to `isPending`, and this will be a breaking change (but we are in v0 so we can do anything anyway).

Reply:
How about `startIfIdle` or `startIfClear` (consistent with `clearTimeout` terminology). Both are short and without negation.

Reply:
Can we have "idle" as a compromise we all seem to like (or at least not hate), then?